### DISCUSSION TITLE
When is MVU coming?

### DISCUSSION THREAD

**Discussion by andrewleader** [dotnet/maui#490] [Q&A]
The roadmap doesn't mention MVU support. Is that coming for Nov 2021 GA release? Or should we not expect that till later? Thanks!

**Comment by davidortinau** [comment]
MVU is an experiment at this point. Follow https://github.com/dotnet/comet for progress.

**Reply by libin85** [reply]
Hopefully it doesn't just stay as an experiment. I personally think it's not a coincidence that frameworks such as  react, flutter, swiftui, fabulous etc all our truly loved by developers. They have one thing in common that's MVU.

We as a community should get behind clancey's effort.

**Reply by qcjxberin** [reply]
If it's just an experiment, I guess I will give up. I have been hesitating between flutter and maui recently. If there is no progress, I think our team will consider using flutter.

**Reply by saint4eva** [reply]
> If it's just an experiment, I guess I will give up. I have been hesitating between flutter and maui recently. If there is no progress, I think our team will consider using flutter.

Blazor started as experiment too, but now it is production ready. I think the Maui team should put in more resources to make sure that Comet MVU is brought over to Maui repo as soon as possible so the community can start contributing. Using C# to write UI is very fundamental to the success of Maui.

**Reply by dansiegel** [reply]
I think it's important to put things in perspective here folks. MVU is a stated goal. We're looking at .NET 6 preview 2 which has the first build of any sort with extremely early bits of what is being call Maui. This isn't close to something you should even be considering for production. 

As of right now it still takes effort to bring that Maui package in from a CI feed. Eventually this will come as part of the build system SDK similar to what you see with AspNetCore. I think if MVU is important to you, it's important to show some patience while the team gets the foundational pieces in place and working. Remember these are EXTREMELY early bits and we still see the team submitting PRs with titles like "Try to get the project to build". 

Want to see it quicker? Look at the issues and chat with the team, see how you can be part of the solution to help further Maui at a quicker pace. Let's flood them with PRs that make this mature quicker, and be the only framework you'd ever consider for building your apps.

**Reply by andrewleader** [reply]
Hmm this is really a bummer, the two reasons I were considering MAUI was (1) single project system rather than the clumsy multi-projects of Xamarin previously, and (2) reactive-based programming (MVU).

I agree with the sentiment here, I wish I could just go with Flutter or React Native for my existing app, but unfortunately I think it'd be too big of a task to convert my code from C#...

Knowing this though, I definitely won't be planning on building a new app on MAUI, I'll move to Flutter/React. Quite a bummer!

Also, in terms of it taking "effort", Microsoft is 1-2 years behind the industry and is going to be 3-4 years behind by the time it seems like something reactive-based will be available (aside from Blazor, which I don't want to use web tech in my mobile/desktop apps). I'm sure Microsoft could throw more resources at this if they saw the value... consider this thread a point of data that could be used to move things along faster rather than asking your customers to build it themselves (otherwise your customers will move to Flutter as people are saying here).

**Reply by qcjxberin** [reply]
Time waits for no one, and the cost of later migration is very high, and the cost of the team is also very high.

**Reply by qcjxberin** [reply]
I am really hesitating at the moment. Once I use flutter, it is basically difficult to come back and use maui again.
